### ISSUE TITLE
Fail to delete a partition from file system

### ISSUE THREAD

**Issue by on99** [crate/crate#6836]
**CrateDB version**: 2.2.5

**JVM version**: openjdk version "1.8.0_151"

**OS version / environment description**: Ubuntu 16.04

**Problem description**: 
Delete a partition in crash/crate admin ui/self-written program in go by `DELETE FROM dsp.trackings WHERE created_at_day = '2018-01-26T00:00:00'` where the table is partitioned by `created_at_day` without any error, the partition is neither queryable nor visible in crate admin ui but disk space in file system is not free.

As there was no error returned, I naively thought the query was executed correctly without any error so I did not even open crate log to check what is going on until now... Here is the crate log.

```
[2018-02-02T00:00:02,660][WARN ][o.e.i.IndicesService     ] [Tête à l'Âne] org.elasticsearch.indices.IndicesService$$Lambda$2426/1535982469@3d4bb0c4
org.elasticsearch.env.ShardLockObtainFailedException: [dsp..partitioned.trackings.04732d9h6osj4d1o60o30c1g][6]: obtaining shard lock timed out after 0ms
        at org.elasticsearch.env.NodeEnvironment$InternalShardLock.acquire(NodeEnvironment.java:726) ~[crate-app-2.2.5.jar:2.2.5]
        at org.elasticsearch.env.NodeEnvironment.shardLock(NodeEnvironment.java:645) ~[crate-app-2.2.5.jar:2.2.5]
        at org.elasticsearch.env.NodeEnvironment.lockAllForIndex(NodeEnvironment.java:591) ~[crate-app-2.2.5.jar:2.2.5]
        at org.elasticsearch.env.NodeEnvironment.deleteIndexDirectorySafe(NodeEnvironment.java:542) ~[crate-app-2.2.5.jar:2.2.5]
        at org.elasticsearch.indices.IndicesService.deleteIndexStoreIfDeletionAllowed(IndicesService.java:653) [crate-app-2.2.5.jar:2.2.5]
        at org.elasticsearch.indices.IndicesService.deleteIndexStore(IndicesService.java:640) [crate-app-2.2.5.jar:2.2.5]
        at org.elasticsearch.indices.IndicesService.removeIndex(IndicesService.java:546) [crate-app-2.2.5.jar:2.2.5]
        at org.elasticsearch.indices.cluster.IndicesClusterStateService.deleteIndices(IndicesClusterStateService.java:258) [crate-app-2.2.5.jar:2.2.5]
        at org.elasticsearch.indices.cluster.IndicesClusterStateService.applyClusterState(IndicesClusterStateService.java:192) [crate-app-2.2.5.jar:2.2.5]
        at org.elasticsearch.cluster.service.ClusterService.callClusterStateAppliers(ClusterService.java:814) [crate-app-2.2.5.jar:2.2.5]
        at org.elasticsearch.cluster.service.ClusterService.publishAndApplyChanges(ClusterService.java:768) [crate-app-2.2.5.jar:2.2.5]
        at org.elasticsearch.cluster.service.ClusterService.runTasks(ClusterService.java:587) [crate-app-2.2.5.jar:2.2.5]
        at org.elasticsearch.cluster.service.ClusterService$ClusterServiceTaskBatcher.run(ClusterService.java:263) [crate-app-2.2.5.jar:2.2.5]
        at org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:150) [crate-app-2.2.5.jar:2.2.5]
        at org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:188) [crate-app-2.2.5.jar:2.2.5]
        at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:569) [crate-app-2.2.5.jar:2.2.5]
        at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:247) [crate-app-2.2.5.jar:2.2.5]
        at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:210) [crate-app-2.2.5.jar:2.2.5]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_151]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_151]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_151]
[2018-02-02T00:12:30,837][INFO ][o.e.m.j.JvmGcMonitorService] [Tête à l'Âne] [gc][305236] overhead, spent [421ms] collecting in the last [1.1s]
[2018-02-02T00:30:02,670][WARN ][o.e.i.c.IndicesClusterStateService] [Tête à l'Âne] [[dsp..partitioned.trackings.04732d9h6osj4d1o60o30c1g/nedbHykWSQipTDYNGc61fA]] failed to complete pending deletion for index
org.elasticsearch.env.ShardLockObtainFailedException: [dsp..partitioned.trackings.04732d9h6osj4d1o60o30c1g][6]: obtaining shard lock timed out after 1800000ms
        at org.elasticsearch.env.NodeEnvironment$InternalShardLock.acquire(NodeEnvironment.java:726) ~[crate-app-2.2.5.jar:2.2.5]
        at org.elasticsearch.env.NodeEnvironment.shardLock(NodeEnvironment.java:645) ~[crate-app-2.2.5.jar:2.2.5]
        at org.elasticsearch.env.NodeEnvironment.lockAllForIndex(NodeEnvironment.java:591) ~[crate-app-2.2.5.jar:2.2.5]
        at org.elasticsearch.indices.IndicesService.processPendingDeletes(IndicesService.java:928) ~[crate-app-2.2.5.jar:2.2.5]
        at org.elasticsearch.indices.cluster.IndicesClusterStateService$2.doRun(IndicesClusterStateService.java:294) [crate-app-2.2.5.jar:2.2.5]
        at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:638) [crate-app-2.2.5.jar:2.2.5]
        at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) [crate-app-2.2.5.jar:2.2.5]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_151]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_151]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_151]
```

**Steps to reproduce**:

```sql
CREATE TABLE IF NOT EXISTS "dsp"."trackings" (
   "created_at" TIMESTAMP,
   "created_at_day" TIMESTAMP GENERATED ALWAYS AS date_trunc('day', "created_at"),
   "duplicate" INTEGER,
   "request_ad_type" STRING,
   "request_ad_unit_id" STRING,
   "request_app_id" STRING,
   "request_bid_id" STRING,
   "request_city" STRING,
   "request_country" STRING,
   "request_idfa" STRING,
   "request_imei" STRING,
   "request_ip" STRING,
   "request_os" STRING,
   "request_region" STRING,
   "request_ssp_id" STRING,
   "response_ad_id" STRING,
   "response_advertiser_id" STRING,
   "response_app_id" STRING,
   "response_creative_id" STRING,
   "response_order_id" STRING,
   "response_pay_type" STRING,
   "response_predicted_cpm" FLOAT,
   "response_transaction_price" FLOAT,
   "response_unit_price" FLOAT,
   "tracking_type" STRING,
   PRIMARY KEY ("tracking_type", "request_bid_id", "created_at_day")
)
CLUSTERED INTO 48 SHARDS
PARTITIONED BY ("created_at_day")
WITH (
   "blocks.metadata" = false,
   "blocks.read" = false,
   "blocks.read_only" = false,
   "blocks.write" = false,
   column_policy = 'strict',
   "mapping.total_fields.limit" = 1000,
   number_of_replicas = '0',
   "recovery.initial_shards" = 'quorum',
   refresh_interval = 1000,
   "routing.allocation.enable" = 'all',
   "routing.allocation.total_shards_per_node" = -1,
   "translog.durability" = 'REQUEST',
   "translog.flush_threshold_size" = 536870912,
   "translog.sync_interval" = 5000,
   "unassigned.node_left.delayed_timeout" = 60000,
   "warmer.enabled" = true,
   "write.wait_for_active_shards" = 'all'
)
```

There are 7 partitions in total, each of them are around 100GB. On 00:00:00 the oldest partition is deleted. Crate cluster is formed by 3 data node and 1 master node.

`df -u` on data nodes (I wipe the irrelevant information)
```
Filesystem      Size  Used Avail Use% Mounted on
/dev/vdb1       126G   97G   24G  81% /data
/dev/vdc1       296G  229G   52G  82% /data2
```

Here is the crate config file, noted that it contains two data path.

```yml
cluster.name: dsp
node.master: false
node.data: true
network.host: _site_
path.data: /data/crate,/data2/crate
bootstrap.memory_lock: true

discovery.zen.minimum_master_nodes: 1
gateway.recover_after_nodes: 3
gateway.expected_nodes: 4
discovery.zen.ping:
  unicast.hosts:
    - master_node_ip:4300

cluster.routing.allocation.disk.watermark.low: 10gb
cluster.routing.allocation.disk.watermark.high: 5gb
```

Please ping me if you want more information as I do not really know the exact steps to reproduce the issue.

**Use case**:

**Feature description**:

**Comment by on99** [comment]
The disk space is almost out for now so I restart crate and hopefully the *deleted* data is truly removed from file system.

I will setup a cronjob to restart crate everyday as a workaround now.

Thank you for help.

**Comment by mxm** [comment]
Thanks for reporting this @on99. The issue here is that the shard file was attempted to be deleted but we couldn't acquire the shard lock within 30 minutes. The shard lock prevents concurrent access on the file. Another operation must have be holding on to the shard lock. Such operations could be a writing/merging results to the shards (translog or index files) or a recovery operation for this shard.

You already discovered that the shard deletion is re-attempted once you recover the index files upon a restart. While the old data won't cause any data corruption in before you restart, you might end up with more disk space occupied. However, we it also try to delete the data again when the next delete operation is scheduled. That means, if you rerun any delete query again, it will re-attempt to perform old delete operations, so the data should eventually be deleted. 

Without further information, it will be very hard to find the cause of this but we will check if other operations might be holding on to the shard lock and not releasing it. Two things that you do:

1) When this happens again, could you take a heap dump of the node where the deletion fails? This would help us finding the operation blocking the shard lock.
2) Could you please try if subsequent delete operations eventually lead to a deletion of the old data?

**Comment by on99** [comment]
@mxm Thank you for your help!

1. Can I know what commands should I execute on the data node? This happens every day. 🤦‍♂️ 
2. My program execute the `DELETE FROM dsp.trackings WHERE created_at_day = '2018-01-26T00:00:00'` query every hour, however, from crate log I can only see the error (lock acquisition) log once a day, all others are info level log about gc issue which are irrelevant. So, to answer your question, subsequent delete operations do not lead to a deletion of the old data, assuming crate is not restarted.

**Comment by on99** [comment]
1. FYI, I executed the same delete query in crash/crate admin ui console for multiple times today, but nothing changes. So my answer to this question remains the same, it does not lead to a deletion of the old data.

**Comment by mxm** [comment]
Heap and thread dump generation:
1. ``PID=`jps -l | grep CrateDB | cut -f1 -d' '` `` should give you the PID of the CrateDB process
2. Use that PID to obtain a thread dump: `jstack -l $PID > thread_dump.txt`
3. Use that PID to obtain a heap dump: `jmap -dump:format=b,file=memory_dump.bin $PID`

>My program execute the DELETE FROM dsp.trackings WHERE created_at_day = '2018-01-26T00:00:00' query every hour, however, from crate log I can only see the error (lock acquisition) log once a day

I've looked into the code again and indeed that the meta data is removed first, then the pending deletes will be scheduled if the shards can't be delete immediately. If this fails after 30 minutes, the index meta data is already gone and it won't be attempted to be deleted again.

Do you have any information on the type of queries you run on the partitions? Are there any queries still running? `Select * from sys.jobs`.

**Comment by on99** [comment]
> Do you have any information on the type of queries you run on the partitions? Are there any queries still running? Select * from sys.jobs.

Yes. There are always some `select` queries touching the pending delete partitions. Does it matter? If so, I can try to modify the `select` queries not to touch those partitions, and see if it happens again.

**Comment by mxm** [comment]
It shouldn't matter but would be interesting to check. Also, check if there are any queries running for a long time in sys.jobs.

**Comment by on99** [comment]
No there is no long time query. All the queries are done in less than 100ms.

**Comment by on99** [comment]
@mxm Sorry the bug does not happen today. I'll email you the heap and thread dump later if I encounter the same issue. This is tricky.

**Comment by mxm** [comment]
Thanks, the dumps would be extremely helpful.

**Comment by on99** [comment]
The issue still does not reproduce.

I just get the thread dump from production server but not sure if it's what you want.
```
Attaching to process ID 30000, please wait...
Debugger attached successfully.
Server compiler detected.
JVM version is 25.151-b12
Deadlock Detection:

java.lang.RuntimeException: Unable to deduce type of thread from address 0x00007f9a20007800 (expected type JavaThread, CompilerThread, ServiceThread, JvmtiAgentThread, or SurrogateLockerThread)
	at sun.jvm.hotspot.runtime.Threads.createJavaThreadWrapper(Threads.java:166)
	at sun.jvm.hotspot.runtime.Threads.first(Threads.java:150)
	at sun.jvm.hotspot.runtime.DeadlockDetector.createThreadTable(DeadlockDetector.java:149)
	at sun.jvm.hotspot.runtime.DeadlockDetector.print(DeadlockDetector.java:56)
	at sun.jvm.hotspot.runtime.DeadlockDetector.print(DeadlockDetector.java:39)
	at sun.jvm.hotspot.tools.StackTrace.run(StackTrace.java:62)
	at sun.jvm.hotspot.tools.StackTrace.run(StackTrace.java:45)
	at sun.jvm.hotspot.tools.JStack.run(JStack.java:66)
	at sun.jvm.hotspot.tools.Tool.startInternal(Tool.java:260)
	at sun.jvm.hotspot.tools.Tool.start(Tool.java:223)
	at sun.jvm.hotspot.tools.Tool.execute(Tool.java:118)
	at sun.jvm.hotspot.tools.JStack.main(JStack.java:92)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at sun.tools.jstack.JStack.runJStackTool(JStack.java:140)
	at sun.tools.jstack.JStack.main(JStack.java:106)
Caused by: sun.jvm.hotspot.types.WrongTypeException: No suitable match for type of address 0x00007f9a20007800
	at sun.jvm.hotspot.runtime.InstanceConstructor.newWrongTypeException(InstanceConstructor.java:62)
	at sun.jvm.hotspot.runtime.VirtualConstructor.instantiateWrapperFor(VirtualConstructor.java:80)
	at sun.jvm.hotspot.runtime.Threads.createJavaThreadWrapper(Threads.java:162)
	... 17 more
Can't print deadlocks:Unable to deduce type of thread from address 0x00007f9a20007800 (expected type JavaThread, CompilerThread, ServiceThread, JvmtiAgentThread, or SurrogateLockerThread)
Exception in thread "main" java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at sun.tools.jstack.JStack.runJStackTool(JStack.java:140)
	at sun.tools.jstack.JStack.main(JStack.java:106)
Caused by: java.lang.InternalError: Metadata does not appear to be polymorphic
	at sun.jvm.hotspot.types.basic.BasicTypeDataBase.findDynamicTypeForAddress(BasicTypeDataBase.java:278)
	at sun.jvm.hotspot.runtime.VirtualBaseConstructor.instantiateWrapperFor(VirtualBaseConstructor.java:102)
	at sun.jvm.hotspot.oops.Metadata.instantiateWrapperFor(Metadata.java:68)
	at sun.jvm.hotspot.memory.DictionaryEntry.klass(DictionaryEntry.java:71)
	at sun.jvm.hotspot.memory.DictionaryEntry.equals(DictionaryEntry.java:79)
	at sun.jvm.hotspot.memory.Dictionary.getEntry(Dictionary.java:99)
	at sun.jvm.hotspot.memory.Dictionary.find(Dictionary.java:87)
	at sun.jvm.hotspot.memory.SystemDictionary.find(SystemDictionary.java:159)
	at sun.jvm.hotspot.memory.SystemDictionary.find(SystemDictionary.java:151)
	at sun.jvm.hotspot.memory.SystemDictionary.getAbstractOwnableSynchronizerKlass(SystemDictionary.java:135)
	at sun.jvm.hotspot.runtime.ConcurrentLocksPrinter.fillLocks(ConcurrentLocksPrinter.java:66)
	at sun.jvm.hotspot.runtime.ConcurrentLocksPrinter.<init>(ConcurrentLocksPrinter.java:36)
	at sun.jvm.hotspot.tools.StackTrace.run(StackTrace.java:71)
	at sun.jvm.hotspot.tools.StackTrace.run(StackTrace.java:45)
	at sun.jvm.hotspot.tools.JStack.run(JStack.java:66)
	at sun.jvm.hotspot.tools.Tool.startInternal(Tool.java:260)
	at sun.jvm.hotspot.tools.Tool.start(Tool.java:223)
	at sun.jvm.hotspot.tools.Tool.execute(Tool.java:118)
	at sun.jvm.hotspot.tools.JStack.main(JStack.java:92)
	... 6 more
```

If it is, I will collect the dump when needed. 🤦‍♂️

**Comment by mxm** [comment]
@on99 I think the issue is that you're not producing the thread dump with the same user that started CrateDB, see also https://bugzilla.redhat.com/show_bug.cgi?id=1095504

**Comment by on99** [comment]
@mxm Thank you for help!

I am sending you the thread dump via email, memory dump is not included because of its size... Again, the issue is not reproduced, just send you the dump to see if it helps. I'll keep an eye on the server log and collect dump once it reproduces.

**Comment by on99** [comment]
@mxm 
I am encountering another issue, that is when I drop the table from crash cli
```sh
drop table dsp.trackings;
```
it did not delete the files from one of the data nodes (out of 3). The `shard lock` log did occur again but I am not sure if it's related. Then I updated crate to 2.3.3 and did a full restart, but no luck, the files are still there which take up most of the disk space of the server.

Log and thread dump is attached here, memory dump is too large (12GB) so I cannot put it here. If you really need the memory dump, please let me know how I can send it you.

[crate.log](https://github.com/crate/crate/files/1754295/yumimobi.log)
[thread_dump.txt](https://github.com/crate/crate/files/1754296/thread_dump_228.txt)

**Comment by mfussenegger** [comment]
It seems like some of the indices have an invalid setting and cannot be read anymore:

```
Caused by: java.lang.IllegalArgumentException: unknown setting [indices.store.throttle.max_byte_per_sec] did you mean [indices.store.throttle.max_byte**s**_per_sec]
```

These were probably created in an earlier version where the setting validation wasn't as strict.
Did you notice on an earlier upgrade that some tables were not loaded? Because this kind of validation isn't new in 2.3 but has been there for a while. At the time of the upgrade this would've been simpler to resolve, now it's more difficult.

```
[2018-02-24T18:45:48,797][WARN ][o.e.c.u.IndexFolderUpgrader] [/data/crate/nodes/0/indices/nedbHykWSQipTDYNGc61fA] no index state found - ignoring
[2018-02-24T18:45:48,797][WARN ][o.e.c.u.IndexFolderUpgrader] [/data2/crate/nodes/0/indices/nedbHykWSQipTDYNGc61fA] no index state found - ignoring
[2018-02-24T18:45:48,828][WARN ][o.e.c.u.IndexFolderUpgrader] [/data/crate/nodes/0/indices/yZgHMAWyQNif7_HYNxoyCw] no index state found - ignoring
[2018-02-24T18:45:48,829][WARN ][o.e.c.u.IndexFolderUpgrader] [/data2/crate/nodes/0/indices/yZgHMAWyQNif7_HYNxoyCw] no index state found - ignoring
[2018-02-24T18:45:48,832][WARN ][o.e.c.u.IndexFolderUpgrader] [/data/crate/nodes/0/indices/xulp55hNRcKI2_2BFTAoQQ] no index state found - ignoring
[2018-02-24T18:45:48,832][WARN ][o.e.c.u.IndexFolderUpgrader] [/data2/crate/nodes/0/indices/xulp55hNRcKI2_2BFTAoQQ] no index state found - ignoring
[2018-02-24T18:45:48,835][WARN ][o.e.c.u.IndexFolderUpgrader] [/data/crate/nodes/0/indices/-QyeNLLHTlOf5b_IXhbSjg] no index state found - ignoring
[2018-02-24T18:45:48,835][WARN ][o.e.c.u.IndexFolderUpgrader] [/data2/crate/nodes/0/indices/-QyeNLLHTlOf5b_IXhbSjg] no index state found - ignoring
[2018-02-24T18:45:48,846][WARN ][o.e.c.u.IndexFolderUpgrader] [/data/crate/nodes/0/indices/BP_t96yxTwC5HL1y1eYAdw] no index state found - ignoring
[2018-02-24T18:45:48,846][WARN ][o.e.c.u.IndexFolderUpgrader] [/data2/crate/nodes/0/indices/BP_t96yxTwC5HL1y1eYAdw] no index state found - ignoring
[2018-02-24T18:45:48,848][WARN ][o.e.c.u.IndexFolderUpgrader] [/data/crate/nodes/0/indices/0uG-WueASka5GNDDCEF3eA] no index state found - ignoring
[2018-02-24T18:45:48,848][WARN ][o.e.c.u.IndexFolderUpgrader] [/data2/crate/nodes/0/indices/0uG-WueASka5GNDDCEF3eA] no index state found - ignoring
[2018-02-24T18:45:48,849][WARN ][o.e.c.u.IndexFolderUpgrader] [/data/crate/nodes/0/indices/NxIPEegoRae-bLE0drP5eQ] no index state found - ignoring
[2018-02-24T18:45:48,849][WARN ][o.e.c.u.IndexFolderUpgrader] [/data2/crate/nodes/0/indices/NxIPEegoRae-bLE0drP5eQ] no index state found - ignoring
[2018-02-24T18:45:48,856][WARN ][o.e.c.u.IndexFolderUpgrader] [/data/crate/nodes/0/indices/SKzwpf3BRBaNq6XPVfgshg] no index state found - ignoring
[2018-02-24T18:45:48,857][WARN ][o.e.c.u.IndexFolderUpgrader] [/data2/crate/nodes/0/indices/SKzwpf3BRBaNq6XPVfgshg] no index state found - ignoring
[2018-02-24T18:45:48,861][WARN ][o.e.c.u.IndexFolderUpgrader] [/data/crate/nodes/0/indices/7zuxQ8bPT86vDqqS0rNeRA] no index state found - ignoring
[2018-02-24T18:45:48,861][WARN ][o.e.c.u.IndexFolderUpgrader] [/data2/crate/nodes/0/indices/7zuxQ8bPT86vDqqS0rNeRA] no index state found - ignoring
[2018-02-24T18:45:48,871][WARN ][o.e.c.u.IndexFolderUpgrader] [/data/crate/nodes/0/indices/dfTYmqyKTQubsalemN8L7Q] no index state found - ignoring
[2018-02-24T18:45:48,872][WARN ][o.e.c.u.IndexFolderUpgrader] [/data2/crate/nodes/0/indices/dfTYmqyKTQubsalemN8L7Q] no index state found - ignoring
[2018-02-24T18:45:48,872][WARN ][o.e.c.u.IndexFolderUpgrader] [/data/crate/nodes/0/indices/le0iYYNVRYiIYENgy7F6Bg] no index state found - ignoring
[2018-02-24T18:45:48,872][WARN ][o.e.c.u.IndexFolderUpgrader] [/data2/crate/nodes/0/indices/le0iYYNVRYiIYENgy7F6Bg] no index state found - ignoring
[2018-02-24T18:45:48,874][WARN ][o.e.c.u.IndexFolderUpgrader] [/data/crate/nodes/0/indices/xPwRGKLFQeC7Z9Hk1AHzOg] no index state found - ignoring
[2018-02-24T18:45:48,874][WARN ][o.e.c.u.IndexFolderUpgrader] [/data2/crate/nodes/0/indices/xPwRGKLFQeC7Z9Hk1AHzOg] no index state found - ignoring
[2018-02-24T18:45:48,879][WARN ][o.e.c.u.IndexFolderUpgrader] [/data/crate/nodes/0/indices/TL74xxmkTnSm04zGSrVWGg] no index state found - ignoring
[2018-02-24T18:45:48,879][WARN ][o.e.c.u.IndexFolderUpgrader] [/data2/crate/nodes/0/indices/TL74xxmkTnSm04zGSrVWGg] no index state found - ignoring
[2018-02-24T18:45:48,881][WARN ][o.e.c.u.IndexFolderUpgrader] [/data/crate/nodes/0/indices/WGMB9qvKSQyMIOx4yMpT_Q] no index state found - ignoring
[2018-02-24T18:45:48,881][WARN ][o.e.c.u.IndexFolderUpgrader] [/data2/crate/nodes/0/indices/WGMB9qvKSQyMIOx4yMpT_Q] no index state found - ignoring
[2018-02-24T18:45:48,882][WARN ][o.e.c.u.IndexFolderUpgrader] [/data/crate/nodes/0/indices/_HIeUbJbSvaLcKk0cbILqA] no index state found - ignoring
[2018-02-24T18:45:48,882][WARN ][o.e.c.u.IndexFolderUpgrader] [/data2/crate/nodes/0/indices/_HIeUbJbSvaLcKk0cbILqA] no index state found - ignoring
[2018-02-24T18:45:48,888][WARN ][o.e.c.u.IndexFolderUpgrader] [/data/crate/nodes/0/indices/viIl_gG5QKWwVr6My4j1Og] no index state found - ignoring
[2018-02-24T18:45:48,889][WARN ][o.e.c.u.IndexFolderUpgrader] [/data2/crate/nodes/0/indices/viIl_gG5QKWwVr6My4j1Og] no index state found - ignoring
[2018-02-24T18:45:48,889][WARN ][o.e.c.u.IndexFolderUpgrader] [/data/crate/nodes/0/indices/BN9y4u4iR62X6Dhu7WsQ_w] no index state found - ignoring
[2018-02-24T18:45:48,889][WARN ][o.e.c.u.IndexFolderUpgrader] [/data2/crate/nodes/0/indices/BN9y4u4iR62X6Dhu7WsQ_w] no index state found - ignoring
[2018-02-24T18:45:48,889][WARN ][o.e.c.u.IndexFolderUpgrader] [/data/crate/nodes/0/indices/WI6Fz-hoQTqLTmurciyPKw] no index state found - ignoring
[2018-02-24T18:45:48,890][WARN ][o.e.c.u.IndexFolderUpgrader] [/data2/crate/nodes/0/indices/WI6Fz-hoQTqLTmurciyPKw] no index state found - ignoring
```

Since the indices are not loaded into the cluster state (could be related to the invalid setting) the files will be kept around. If you are sure that these are the deleted partitions you could simply delete them directly on the filesystem.

As I said, fixing the setting error is more difficult. If this occurs during a upgrade usually one can roll back, fix the error and then upgrade. But I think this would now involve a major version downgrade which is not supported. We could probably come up with a way to resolve this manually, but that is out of scope of the kind of support we can provide for open source users. 
Depending on what kind of downtime you can afford, the simplest way would be to backup the tables you require and then wipe the cluster and re-import the data.

The original shard-lock issue is of course something we want and need to fix. It seems to be related to https://github.com/crate/crate/issues/6887 and there have already been 2 fixes in `2.3.3` which were identified while investigating the issue.

**Comment by on99** [comment]
@mfussenegger 

BTW, the config `indices.store.throttle.max_byte_per_sec: 200mb` was added after upgrading to 2.3.3 but the issue that files not deleted from fs happen before upgrade.

I noticed the invalid config once cluster was upgraded and restarted so I deleted that line of config and restarted the cluster. However, the files were still kept in the file system.

And then I noticed that metabase cannot work with crate 2.3.3 #6976 so I decided to downgrade to 2.2.5 which was the original version of crate and restart the cluster, however, the index files of `dropped` table were still there.

Not really know how I can get rid of all those files from fs.
